upup/pkg/fi/cloudup/awstasks/launchtemplate_target_terraform.go

```diff
@@ -173,6 +182,10 @@ func (t *LaunchTemplate) RenderTerraform(target *terraform.TerraformTarget, a, e
 		ImageID:      image,
 		InstanceType: e.InstanceType,
 		Lifecycle:    &terraform.Lifecycle{CreateBeforeDestroy: fi.Bool(true)},
+		MetadataOptions: &terraformLaunchTemplateInstanceMetadata{
+			HTTPTokens:              e.HTTPTokens,
+			HTTPPutResponseHopLimit: e.HTTPPutResponseHopLimit,
+		},
```

Comment:
[I've found](https://kubernetes.slack.com/archives/C8MKE2G5P/p1607461310392500?thread_ts=1607120672.355200&cid=C8MKE2G5P) that this breaks using Terraform 0.12.26, with an error message like this:
```
Error: error creating Launch Template (lt-0f27a1c3cb68aba0f) Version: InvalidParameterValue: A value of ‘’ is not valid for http-endpoint. Valid values are ‘enabled’ or ‘disabled’.
	status code: 400, request id: fd0c6d8c-ad70-4877-9adf-00e6e35f2edc

  on kubernetes.tf line 695, in resource "aws_launch_template" "bastions-redacted":
 695: resource "aws_launch_template" "bastions-redacted" {
```
There's an open Terraform issue describing this problem: hashicorp/terraform-provider-aws#12564. If I add [the "http_endpoint" attribute](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#http_endpoint) in manually, it works. The documentation for that attribute says that it's optional with a reasonable default value, but in practice either Terraform is not sending a value at all, or it's sending the wrong value.

Reply:
Using terraform is such a joy...

Reply:
Please see #10393.